src/klein/_decorators.py

```diff
@@ -20,11 +24,16 @@ def bindable(bindable):
     @return: its argument, modified to mark it as unconditinally requiring an
         instance argument.
     """
-    bindable.__klein_bound__ = True
+    bindable.__klein_bound__ = True  # type: ignore[attr-defined]
     return bindable
 
 
-def modified(modification, original, modifier=None):
+def modified(
+    modification,  # type: Text
+    original,  # type: Callable
+    modifier=None,  # type: Optional[Callable]
+):
+    # type: (...) -> Callable
```

Comment:
If C2 is a new `TypeVar`, I think we can give this a more precise signature, like this:

```suggestion
def modified(
    modification,  # type: Text
    original,  # type: C
    modifier=lambda x: x,  # type: Callable[[C], C2]
):
    # type: (...) -> C2
```

but if this provokes more mypy issues, feel free to defer it.

Reply:
Yeah this ends up being a little tricky.

Reply:
Filed as #345